docs/ADR/001-porting-to-postgresql.md

```diff
@@ -0,0 +1,10 @@
+# ADR 001 - How we migrate Router/Router-API from Mongo to Postgres
+
+* Spin up a completely separate stack of router and router-api (govuk-docker first) that will talk to a PostgreSQL db only (as forks)
+* Provision in EKS a new PostgreSQL db
+* Provision the new p-router/p-router-api apps
+* Amend messages to router-api to be duplicated to p-router-api within the gds-api-adapters gem
```

Comment:
This should be in content-store. The gds-api-adapters gem is just helper methods, it shouldn't have "business" logic in it.

Reply:
So I'd split this into two steps. One to add helper methods to gds-api-adapter to call `p-router-api` (this still needs a better name!) and the next step to call the new helper methods from content store.

Reply:
We did consider doing it in content-store, but felt it much easier to implement and manage (temporarily) in the gem.

Good point on the splitting of steps.

Reply:
@chao-xian the suggestion from me to do the logic in the API (as small as that logic is) was based on the theory that it's one solid point of entry for everything. @leenagupte if content store is the only location that actually accesses these helper calls, we could do it there.

However to clarify, there is no business logic as such in the API adaptor in this plan, we simply append a call to another helper method inside the existing one (as is the plan).

Reply:
Content Store is the only place that calls router / router-api. You'd be making a bit of a mess in gds-api-adapters if you try to have the router helpers call two places as you'd need to some work to get it to figure out what the endpoint is.

Reply:
My opinion is that either the API or the content store way of doing it would be fine, but the question of where/when to report errors probably makes the decision. Basically, if we do this splitting we need to be able to know about the following error cases:

-